### ISSUE TITLE
llvm-symbolizer should "Just Work"

### ISSUE THREAD

**Issue by ramosian-glider** [ramosian-glider/sanitizers#135]
Originally reported on Google Code with ID 134

```
(see also https://codereview.chromium.org/11348066/)
The users expect ASan to print understandable error logs without any additional env
or output filtering.
Can we hardcode the compiler path somewhere so that ASan looks it up for llvm-symbolizer
if the path hasn't been supplied via env?
```

Reported by `ramosian.glider` on 2012-12-11 14:48:18

**Comment by ramosian-glider** [comment]
```
>> Can we hardcode the compiler path
Instead, we should defeat the symbolizer to make it in-process. 
```

Reported by `konstantin.s.serebryany` on 2012-12-11 14:55:11

**Comment by ramosian-glider** [comment]
```
Please remind me, have we considered putting the symbolizer into a .so?
```

Reported by `ramosian.glider` on 2012-12-11 15:21:58

**Comment by ramosian-glider** [comment]
```
Loading a .so with the system loader after a memory error or SIGSEGV occurs seems dicey.
 Couldn't you be inside malloc or the loader?

If you ignore stacks from signal handlers, then maybe the problem is easier.  From
signal handlers, it seems like the symbolizer has to be careful to avoid malloc.
```

Reported by `rnk@google.com` on 2012-12-11 15:34:10

**Comment by ramosian-glider** [comment]
Reported by `konstantin.s.serebryany` on 2013-02-15 14:28:44
- **Status changed**: `Duplicate`
- **Merged into**: #16

**Comment by ramosian-glider** [comment]
```
Adding Project:AddressSanitizer as part of GitHub migration.
```

Reported by `ramosian.glider` on 2015-07-30 09:13:40
- **Labels added**: ProjectAddressSanitizer
